server/src/main/java/org/apache/druid/query/ResultLevelCachingQueryRunner.java

```diff
@@ -106,6 +106,8 @@ public Sequence<T> run(QueryPlus queryPlus, ResponseContext responseContext)
 
       if (useResultCache && newResultSetId != null && newResultSetId.equals(existingResultSetId)) {
         log.debug("Return cached result set as there is no change in identifiers for query %s ", query.getId());
+        // Call accumulate on the sequence to ensure that all Wrapper/Closer/Baggage/etc. get called
+        resultFromClient.accumulate(null, (accumulated, in) -> accumulated);
```

Comment:
its quite unfortunate that the seq needs to be created to get `newResultSetId` as that is passed thru the responseContext.

actually the returned seq is an empty seq in cases when the cache supposed to be used.